src/guid.c

```diff
@@ -45,29 +45,12 @@ efi_guid_to_str(const efi_guid_t *guid, char **sp)
 	int rc = -1;
 
 	if (!sp) {
-		rc = snprintf(NULL, 0, GUID_FORMAT,
-				le32_to_cpu(guid->a),
-				le16_to_cpu(guid->b),
-				le16_to_cpu(guid->c),
-				be16_to_cpu(guid->d),
-				guid->e[0], guid->e[1], guid->e[2], guid->e[3],
-				guid->e[4], guid->e[5]);
+		rc = snprintf(NULL, 0, GUID_FORMAT, GUID_FORMAT_ARGS(guid));
```

Comment:
AFAICS it was wrong the first time: in all cases everything except guid->d should be stored in memory in host endian, and guid->d should always be stored the opposite.  If we were reading/writing from a binary on disk, we would have to fix them up to make that true.

But also this rarely gets run on BE platforms because EFI doesn't support them.